app/src/main/java/org/schabi/newpipe/util/NavigationHelper.java

```diff
@@ -78,6 +74,9 @@ public static Intent getPlayerIntent(@NonNull final Context context,
         if (quality != null) intent.putExtra(VideoPlayer.PLAYBACK_QUALITY, quality);
         intent.putExtra(VideoPlayer.RESUME_PLAYBACK, resumePlayback);
 
+        int playerType = intent.getIntExtra(VideoPlayer.PLAYER_TYPE, VideoPlayer.PLAYER_TYPE_VIDEO);
```

Comment:
Why are you retrieving data from a newly generated intent? Also, I saw this function is often followed by a `putExtra(VideoPlayer.PLAYER_TYPE`, can't `playerType` be made an argument?

Reply:
> Why are you retrieving data from a newly generated intent? 

Have noooo idea...

> can't playerType be made an argument?

I tried, but it would require to place a value for this argument in sooo many places because getPlayerIntent is used many times across the app. It's better to have 5 placed where PLAYER_TYPE is set inside NavigationHelper.

Reply:
> I tried, but it would require to place a value for this argument in sooo many places because getPlayerIntent is used many times across the app. It's better to have 5 placed where PLAYER_TYPE is set inside NavigationHelper.

That's right, so maybe let's not do this for now ;-)
Is the result any different if you replace lines 77 to 79 with `intent.putExtra(PLAYER_TYPE, PLAYER_TYPE_VIDEO)`?

Reply:
> I tried, but it would require to place a value for this argument in sooo many places because getPlayerIntent is used many times across the app. It's better to have 5 placed where PLAYER_TYPE is set inside NavigationHelper.

That's right, so maybe let's not do this for now ;-)
Is the result any different if you replace lines 77 to 79 with `intent.putExtra(PLAYER_TYPE, PLAYER_TYPE_VIDEO)`?

Reply:
That would be exactly the same, as getting something that isn't set yet with a fallback, will always result in that fallback, and then setting it to that, will thus always result into setting it to that fallback. (Btw I removed your double comment.)